src/pudl/output/denorm_eia860.py

```diff
@@ -0,0 +1,55 @@
+"""Denormalized versions of the EIA 860 tables."""
+import pandas as pd
+from dagster import asset
+
+import pudl
+
+
+@asset(io_manager_key="pudl_sqlite_io_manager", compute_kind="Python")
+def denorm_ownership_eia860(
```

Comment:
non-blocking: Not sure if you've already discussed this somewhere else, but:

If you're looking to reduce the risk of divergent logic implementations, you could split the computation logic into a function that gets called both here and in `output/eia860.py`... something like

```
@asset(io_manager_key="pudl_sqlite_io_manager", compute_kind="Python")
def denorm_ownership_eia860(
    denorm_plants_utilities_eia: pd.DataFrame,
    ownership_eia860: pd.DataFrame,
) -> pd.DataFrame:
    """A denormalized version of the EIA 860 ownership table.

    TODO: Convert to SQL view?

    Args:
        denorm_plants_utilities_eia: Denormalized table containing plant and utility
            names and IDs.
        ownership_eia860: EIA 860 ownership table.

    Returns:
        A denormalized version of the EIA 860 ownership table.
    """

    return generate_denorm_ownership_eia_860(denorm_plants_utilities_eia, ownership_eia860)


def generate_denorm_ownership_eia860(denorm_plants_utilities_eia, ownership_eia860):
    pu_df = denorm_plants_utilities_eia.loc[
        :,
        [
            "plant_id_eia",
            "plant_id_pudl",
            "plant_name_eia",
            "utility_name_eia",
            "utility_id_pudl",
            "report_date",
        ],
    ]
    own_df = pd.merge(
        ownership_eia860, pu_df, on=["report_date", "plant_id_eia"], how="left"
    ).dropna(
        subset=["report_date", "plant_id_eia", "generator_id", "owner_utility_id_eia"]
    )
    first_cols = [
        "report_date",
        "plant_id_eia",
        "plant_id_pudl",
        "plant_name_eia",
        "utility_id_eia",
        "utility_id_pudl",
        "utility_name_eia",
        "generator_id",
        "owner_utility_id_eia",
        "owner_name",
    ]

    # Re-arrange the columns for easier readability:
    return pudl.helpers.organize_cols(own_df, first_cols)
```

And then in the `output/eia860.py`:

```
def ownership_eia860(pudl_engine, start_date=None, end_date=None):
    """Pull a useful set of fields related to ownership_eia860 table.

    Args:
        pudl_engine (sqlalchemy.engine.Engine): SQLAlchemy connection engine
            for the PUDL DB.
        start_date (date-like): date-like object, including a string of the
            form 'YYYY-MM-DD' which will be used to specify the date range of
            records to be pulled.  Dates are inclusive.
        end_date (date-like): date-like object, including a string of the
            form 'YYYY-MM-DD' which will be used to specify the date range of
            records to be pulled.  Dates are inclusive.

    Returns:
        pandas.DataFrame: A DataFrame containing a useful set of fields related
        to the EIA 860 Ownership table.
    """
    pt = pudl.output.pudltabl.get_table_meta(pudl_engine)
    own_eia860_tbl = pt["ownership_eia860"]
    own_eia860_select = sa.sql.select(own_eia860_tbl)

    if start_date is not None:
        start_date = pd.to_datetime(start_date)
        own_eia860_select = own_eia860_select.where(
            own_eia860_tbl.c.report_date >= start_date
        )
    if end_date is not None:
        end_date = pd.to_datetime(end_date)
        own_eia860_select = own_eia860_select.where(
            own_eia860_tbl.c.report_date <= end_date
        )
    own_eia860_df = pd.read_sql(own_eia860_select, pudl_engine).assign(
        report_date=lambda x: pd.to_datetime(x["report_date"])
    )

    denorm_plants_utilities_eia = plants_utils_eia860(
        pudl_engine, start_date=start_date, end_date=end_date
    )
    out_df = generate_denorm_ownership_eia860(denorm_plants_utilities_eia, own_eia860_df)
    return out_df
```


They could also all be in the same file, even - the kernel, plus the two different wrappers that feed it the right dataframes - though I'm not familiar enough with the various access patterns to judge whether that'd be better or worse...